### PR TITLE
Add tests to verify fix for issue #26854 and add transactional resource function test

### PR DESCRIPTION
**Pull request by lasinicl**
related issue --> https://github.com/ballerina-platform/ballerina-lang/issues/26854
fix --> https://github.com/ballerina-platform/ballerina-lang/pull/27339
(Should merge this after merging the above fix)

### CODE REVIEW

File: transaction-ballerina/src/transaction/tests/transaction_stmt.bal [hunk 1]
```diff
@@ -512,3 +512,56 @@ function testIsolatedTransactionalAnonFunc() {
...
     test:assertEquals(ss, "trxStarted -> within isolated transactional anon func -> trxEnded.");
 }
+
+@test:Config {
+}
+function testRollbackInElseStatement() returns error? {
+    string ss = "";
+    transaction {
+        ss = "trxStarted";
+        var x = commit;
```
Comment (pcnfernando) [inline transaction-ballerina/src/transaction/tests/transaction_stmt.bal:522]: This should not be allowed as per spec. Once we do a commit, the strand comes out of transactional mode. And for a rollback to be allowed, the strand should be in transactional mode.
The following is allowed:
```ballerina
if(something) {
  commit;
} else {
  rollback;
}
```

Comment (lasinicl) [reply transaction-ballerina/src/transaction/tests/transaction_stmt.bal]: Fixed

**Comment by gimantha** [comment]
@lasinicl Shall we update this PR?

**Comment by lasinicl** [comment]
Closing this since the tests are not valid.

**Pull request closed without merging**